airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java

```diff
@@ -219,42 +219,27 @@ private String mapGenre(String genre) {
      * track numbers on the form "4/12".
      */
     private Integer parseTrackNumber(String trackNumber) {
-        if (trackNumber == null) {
-            return null;
-        }
-
-        Integer result = null;
-
-        try {
-            result = Integer.valueOf(trackNumber);
-        } catch (NumberFormatException x) {
-            Matcher matcher = TRACK_NUMBER_PATTERN.matcher(trackNumber);
-            if (matcher.matches()) {
-                try {
-                    result = Integer.valueOf(matcher.group(1));
-                } catch (NumberFormatException e) {
-                    return null;
-                }
-            }
-        }
-
-        if (Integer.valueOf(0).equals(result)) {
-            return null;
-        }
-        return result;
+        return parseIntegerPattern(trackNumber, TRACK_NUMBER_PATTERN);
     }
 
     private Integer parseYear(String year) {
-        if (year == null) {
+        return parseIntegerPattern(year, YEAR_NUMBER_PATTERN);
+    }
+
+    private Integer parseIntegerPattern(String str, Pattern pattern) {
+        if (str == null) {
             return null;
         }
 
         Integer result = null;
 
         try {
-            result = Integer.valueOf(year);
+            result = Integer.valueOf(str);
         } catch (NumberFormatException x) {
-            Matcher matcher = YEAR_NUMBER_PATTERN.matcher(year);
+            if (pattern == null) {
+                return null;
+            }
+            Matcher matcher = pattern.matcher(str);
             if (matcher.matches()) {
                 try {
```

Comment:
Please refactor this a bit to avoid nested exceptions ♥

Reply:
For this pass I was only removing duplicated code.  Refactoring can be done in another PR if desired.